### PR TITLE
fix: avoid NPE when processing not parameterized CRs

### LINKED ISSUE #2905: NPE when attempting to generate non-parameterized CustomResources

### PR DESCRIPTION
**Pull request by metacosm**
Fixes #2905

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
E2E tests are failing due to `kubenetes-itests/` pom not being updated as per CRD generator module changes:

https://github.com/fabric8io/kubernetes-client/blob/fced0b77bf0be852593e0afb332ff40036cc000d/kubernetes-itests/pom.xml#L39-L42

**Review by rohanKanojia — APPROVED** [review]

File: kubernetes-itests/pom.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
     </dependency>
     <dependency>
       <groupId>io.fabric8</groupId>
-      <artifactId>crd-generator</artifactId>
+      <artifactId>crd-generator-apt</artifactId>
```
Comment (jorsol) [inline kubernetes-itests/pom.xml:41]: Is this dependency need in `kubernetes-itest`? I don't see where or how it's used.

Comment (rohanKanojia) [reply kubernetes-itests/pom.xml:41]: I think it's required for CRD Generator tests

https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-itests/src/test/java/io/fabric8/crd/zookeeper/ZookeeperCustomResourceIT.java

Comment (metacosm) [reply kubernetes-itests/pom.xml:41]: Yes, this contains an annotation processor which generates the CRDs from the POJOs. It's not used in the code itself.

Comment (jorsol) [reply kubernetes-itests/pom.xml:41]: Ok, I see, but I will need to open an issue about this.

**Comment by metacosm** [comment]
Superseded by #2909

**Pull request closed without merging**